rasterio/_io.pyx

```diff
@@ -1150,7 +1106,7 @@ cdef class DatasetWriterBase(DatasetReaderBase):
 
         if mode in ('w', 'w+'):
 
-            _delete_dataset_if_exists(path)
+            delete(path, driver)
```

Comment:
@normanb I like this idea, but it's not working: the given driver is often unable to read the dataset to find all of its auxiliary files. Thus the errors like this one:

https://travis-ci.org/github/mapbox/rasterio/jobs/676036878#L3440-L3457